gateway/src/services/config-manager-v2.ts

```diff
@@ -3,29 +3,53 @@ import { ValidateFunction } from 'ajv';
 import fs from 'fs';
 import path from 'path';
 import yaml from 'js-yaml';
+import * as migrations from '../../conf/migration/migrations';
 
 type Configuration = { [key: string]: any };
 type ConfigurationDefaults = { [namespaceId: string]: Configuration };
+type Migration = () => void;
+type MigrationFunctions = {
+  [key: string]: Migration;
+};
 interface _ConfigurationNamespaceDefinition {
   configurationPath: string;
   schemaPath: string;
 }
 type ConfigurationNamespaceDefinition = _ConfigurationNamespaceDefinition & {
   [key: string]: string;
 };
-type ConfigurationRoot = {
+type ConfigurationPattern = {
   [namespaceId: string]: ConfigurationNamespaceDefinition;
 };
+type ConfigurationRoot = {
```

Comment:
Follow the convention we have... if this has fixed members, then it's an interface.